backend/services/implementations/testSessionService.ts

```diff
@@ -189,6 +204,69 @@ class TestSessionService implements ITestSessionService {
 
     return testSessionDtos;
   }
+
+  /*
+   * computeTestGrades computes the breakdown and score of a given
+   * UngradedResultDTO and returns the GradedResultResponseDTO
+   */
+  async computeTestGrades(
+    result: UngradedResultDTO,
+    testId: string,
+  ): Promise<GradedResultResponseDTO> {
+    let gradedResultResponseDTO: GradedResultResponseDTO;
+
+    // the list of a student's answers with each field being either the
+    // numeric answer (for short answer) or index (for multiple choice)
+    const studentAnswers: (number | null)[] = result.answers;
+
+    let computedScore = 0.00;
+    const computedBreakdown: boolean[] = [];
+    let questionsCorrect = 0;
+
+    try {
+      const test: TestResponseDTO = await this.testService.getTestById(testId);
+
+      test.questions.forEach((question: Question, i) => {
+        let actualAnswer: number;
+
+        if (question.questionType === QuestionType.MULTIPLE_CHOICE) {
+          const questionMetadata = question.questionMetadata as MultipleChoiceMetadata;
+          actualAnswer = questionMetadata.answerIndex;
+        } else {
```

Comment:
can we switch this to an `else if`? just in case we decide to introduce another question type. Another option is to use a switch statement

Reply:
switch won't work cause `QuestionType.MULTIPLE_CHOICE=0` always evaluates to true :(. i wonder if there's a way to make it work... but will use if statement for simplicity

Reply:
I think it's possible - see [here](https://stackoverflow.com/questions/50774790/switch-for-specific-type-in-typescript). I'll leave it up to you if you want to change it